nexus/db-queries/src/db/queries/next_item.rs

```diff
@@ -81,23 +82,24 @@ use uuid::Uuid;
 /// FROM
 ///     (
 ///         SELECT
-///             shift
+///             index, shift
 ///         FROM
-///             generate_series(0, <max_shift>)
-///         AS shift
+///             generate_series(0, <max_shift>) AS shift,
+///             generate_series(0, <n>) AS index
 ///         UNION ALL
 ///         SELECT
 ///             shift
```

Comment:
nit: `index, shift`?

Reply:
Done, here's what we get now, as seen by the DTrace query-start probe:

```
INSERT INTO "test_schema"."item" ("id", "value", "time_deleted") VALUES (gen_random_uuid(), (SELECT $1 + "shift" AS "value" FROM (SELECT generate_series(0, $2) AS "index", generate_series(0, $3) AS "shift" UNION ALL SELECT generate_series($4, $5) AS "index", generate_series($6, -1) AS "shift") LEFT OUTER JOIN "test_schema"."item" ON ("value", "time_deleted" IS NULL) = ($7 + "shift", TRUE)  WHERE "value" IS NULL ORDER BY "index" LIMIT 1), NULL) RETURNING "test_schema"."item"."id", "test_schema"."item"."value", "test_schema"."item"."time_deleted" -- binds: [5, 5, 5, 6, 10, -5, 5]
```